setPedStat for client side peds

ArranTuna opened the pull request.

ArranTuna committed:
setPedStat for client side peds

jushar commented on Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp:
```diff
@@ -2077,6 +2079,57 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
+    if ( !argStream.HasErrors () )
+    {
+        // Valid ped?
+        if ( pPed )
```
This line is not necessary as the error case is already handled by `CScriptArgReader::HasErrors`

jushar commented on Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp:
```diff
@@ -2077,6 +2079,57 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
+                m_pScriptDebugging->LogCustom ( luaVM, argStream.GetFullErrorMessage ( ) );
+        }
+        else
+            m_pScriptDebugging->LogBadPointer ( luaVM, "ped", 1 );
```
This can be removed then.

jushar reviewed (commented):
Looks good all in all.

In addition to the in-code comments, you could avoid the double `m_pScriptDebugging->LogCustom(...)` by something like this:
```
// ...
    if (!argStream.HasErrors())
    {
        pPed->SetStat(usStat, fValue);
        // ...
    }
}

// at the end:
if (argStream.HasErrors())
    m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
```

ArranTuna committed:
Improved setPedStat for client side peds.

Implemented Jusonex's feedback, fixed not checking if stat was < 0 and
changed message about max stat from 231 (highest on MTA wiki) to 343
which is what NUM_PLAYER_STATS is.

ArranTuna commented:
Thanks for the review, I've made a new commit. About that if ( pPed ) I assumed that it was needed as many functions in that file have those checks, like getPedStat does. If these checks are useless then can I remove them all in a new patch?

ArranTuna commented:
I guess something like this needs adding:
MinClientReqCheck ( argStream, MIN_CLIENT_REQ_REMOVEPEDFROMVEHICLE_CLIENTSIDE, "function is being called client side" );

ArranTuna committed:
Merge remote-tracking branch 'refs/remotes/multitheftauto/master'

ArranTuna commented:
I seem to have messed up by using the master branch for a pull request so going to close this and make a new one that uses a branch.
